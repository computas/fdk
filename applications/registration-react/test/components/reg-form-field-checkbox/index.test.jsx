import React from 'react';
import { shallow } from 'enzyme';
import CheckboxField from '../../../src/components/reg-form-field-checkbox';

let defaultProps;
let wrapper;
let preventDefault;
let onChange;

beforeEach(() => {
  preventDefault = jest.fn();
  onChange = jest.fn();
  defaultProps = {
    input: { name: 'language', value: [], onChange },
    label: 'label'
  };
  wrapper = shallow(<CheckboxField {...defaultProps} />);
});

test('should render CheckboxField correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle uncheck NOR', () => {
  const mockedEvent = {
    target: { value: 'NOR' },
    preventDefault
  };
  wrapper
    .find('input')
    .first()
    .prop('onChange')(mockedEvent);
  expect(onChange).toHaveBeenCalled();
});

test('should handle check NOR', () => {
  const mockedEvent = {
    target: { value: 'NOR', checked: 'checked' },
    preventDefault
  };
  wrapper
    .find('input')
    .first()
    .prop('onChange')(mockedEvent);
  expect(onChange).toHaveBeenCalled();
});

test('should handle check ENG', () => {
  const mockedEvent = {
    target: { value: 'ENG', checked: 'checked' },
    preventDefault
  };
  wrapper
    .find('input')
    .first()
    .prop('onChange')(mockedEvent);
  expect(onChange).toHaveBeenCalled();
});

test('should handle check SMI', () => {
  const mockedEvent = {
    target: { value: 'SMI', checked: 'checked' },
    preventDefault
  };
  wrapper
    .find('input')
    .first()
    .prop('onChange')(mockedEvent);
  expect(onChange).toHaveBeenCalled();
});
