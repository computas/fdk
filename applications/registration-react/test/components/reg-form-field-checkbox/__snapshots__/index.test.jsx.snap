// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckboxField correctly 1`] = `
<div>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="ENG"
  >
    <input
      checked=""
      id="ENG"
      name="language"
      onChange={[Function]}
      type="checkbox"
      value="ENG"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Engelsk
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="NOR"
  >
    <input
      checked=""
      id="NOR"
      name="language"
      onChange={[Function]}
      type="checkbox"
      value="NOR"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Norsk
    </span>
  </label>
  <label
    className="form-check fdk-form-checkbox"
    htmlFor="SMI"
  >
    <input
      checked=""
      id="SMI"
      name="language"
      onChange={[Function]}
      type="checkbox"
      value="SMI"
    />
    <span
      className="form-check-label fdk-form-check-label"
    />
    <span>
      Samisk
    </span>
  </label>
</div>
`;
